support/netutil/netutil.go

```diff
@@ -72,3 +72,64 @@ func watchSignalWithinTimeout(timeout time.Duration, errSignal chan struct{}, ac
 	}
 	action()
 }
+
+// get the local area network IP address
+func GetLocalHost() string {
+	addresses, err := net.InterfaceAddrs()
+	if err != nil {
+		fmt.Printf("net.InterfaceAddrs failed, err: %v", err.Error())
+		return ""
+	}
+
+	for _, address := range addresses {
+		if ipNet, ok := address.(*net.IPNet); ok && !ipNet.IP.IsLoopback() {
+			// IPv4
+			if ipNet.IP.To4() != nil {
+				return ipNet.IP.String()
+			}
+			// IPv6
+			if ipNet.IP.To16() != nil && ipNet.IP.To4() == nil {
+				return ipNet.IP.String()
+			}
+		}
+	}
+	fmt.Printf("no network interface found")
```

Comment:
This one may cause problem

If this method failed to find an IP, you will display an error message, but the user may have provided 0.0.0.0.

So you will use it, but an error will be displayed

It's unlikely to happen I think, but I would expect the local IPs to be tested after the valid one would be tested first